Sources/SwiftParser/Expressions.swift

```diff
@@ -2088,14 +2093,33 @@ extension Parser {
 // MARK: Conditional Expressions
 
 extension Parser {
+  
+  func extractCodeBlockFromLast(_ conditions: inout RawConditionElementListSyntax) -> RawCodeBlockSyntax? {
+    if let closure = conditions.elements.last?.condition.as(RawClosureExprSyntax.self), closure.signature == nil {
+      var elements = conditions.elements
+      elements.removeLast()
+      conditions = RawConditionElementListSyntax(elements: elements, arena: self.arena)
+      return RawCodeBlockSyntax(
+        closure.unexpectedBeforeLeftBrace,
+        leftBrace: closure.leftBrace,
+        statements: closure.statements,
+        closure.unexpectedBetweenStatementsAndRightBrace,
+        rightBrace: closure.rightBrace,
+        closure.unexpectedAfterRightBrace,
+        arena: self.arena
+      )
+    } else {
+      return nil
+    }
+  }
```

Comment:
I would prefer if we didn’t have to do this. Because all the syntax nodes are allocated in a `SyntaxArena` and not freed until the arena gets deallocated, the original `RawConditionElementListSyntax` will continue to stick around and use memory. 

We sometimes modify already parsed syntax nodes in case there is a syntax error but in that case we say that using a little memory is OK because source files usually don’t have many errors. But in valid code, I would prefer to avoid it. It probably means that you need to implement some lookahead logic to determine whether an item is a closure or the if body.